create-plugins-list-file.sh

```
@@ -36,6 +36,30 @@ get_plugin_name() {
+    fi
+    if [ -z "$java_version" ]; then
+        java_version=$(xmllint --xpath "string(//plugin[artifactId='maven-compiler-plugin']/configuration/target)" "$pom_file")
+    fi
+    echo "$java_version"
+}
+
+# Function to extract Jenkins core version from pom.xml
+# Function to extract Jenkins core version from pom.xml
+get_jenkins_core_version_from_pom() {
+    local pom_file=$1
+    local jenkins_core_version=$(xmllint --xpath "string($pom_xml_jenkins_core_version_xpath)" "$pom_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)